crates/pathfinder/src/p2p_network/client.rs

```diff
@@ -0,0 +1,405 @@
+                                    },
+                                )
+                            },
+                            n_steps: common.execution_resources.n_steps,
+                            n_memory_holes: common.execution_resources.n_memory_holes,
+                        }),
+                        l1_to_l2_consumed_message: common.consumed_message.map(|x| {
+                            gw::L1ToL2Message {
+                                from_address: EthereumAddress(x.from_address),
+                                payload: x
+                                    .payload
+                                    .into_iter()
+                                    .map(L1ToL2MessagePayloadElem)
+                                    .collect(),
+                                selector: EntryPoint(x.entry_point_selector),
+                                to_address: ContractAddress::new_or_panic(x.to_address),
+                                nonce: Some(L1ToL2MessageNonce(x.nonce)),
+                            }
+                        }),
+                        l2_to_l1_messages: common
+                            .messages_sent
+                            .into_iter()
+                            .map(|m| gw::L2ToL1Message {
+                                from_address: ContractAddress::new_or_panic(m.from_address),
+                                payload: m
+                                    .payload
+                                    .into_iter()
+                                    .map(L2ToL1MessagePayloadElem)
+                                    .collect(),
+                                to_address: EthereumAddress(m.to_address),
+                            })
+                            .collect(),
+                        transaction_hash: TransactionHash(common.transaction_hash),
+                        transaction_index: TransactionIndex::new_or_panic(
+                            common.transaction_index.into(),
+                        ),
+                    },
+                }
+            }
+        }
+    }
+
+    pub mod state_diff {
+        use std::collections::HashMap;
+
+        use p2p_proto::sync::BlockStateUpdateWithHash;
+        use pathfinder_common::{
+            BlockHash, CasmHash, ClassHash, ContractAddress, ContractNonce, SierraHash,
+            StorageAddress, StorageValue,
+        };
+        use starknet_gateway_types::reply::state_update as gw;
+
+        pub fn from_p2p(su: BlockStateUpdateWithHash) -> (BlockHash, gw::StateDiff) {
+            (
+                BlockHash(su.block_hash),
+                gw::StateDiff {
+                    storage_diffs: su
+                        .state_update
+                        .contract_diffs
+                        .iter()
+                        .map(|contract_diff| {
+                            (
+                                ContractAddress::new_or_panic(contract_diff.contract_address),
```

Comment:
Aren't all of these `new_or_panic` bugs? They come in as `Felt` from what I can tell.

Reply:
Yeah, good catch, thanks!